woodwork/type_sys/type_system.py

```diff
@@ -261,6 +261,10 @@ def infer_logical_type(self, series):
         if ks and isinstance(series, ks.Series):
             series = series.head(100000).to_pandas()
 
+        # Special case: if the entire column is Null or NaN, use the Unknown logical type
+        if all(series.isnull()):
```

Comment:
does this catch the column of `'nan'` strings? Or are we going to handle that in a separate PR?

Reply:
It only catches true null values, not the string nan representations. That's going to get handled in a separate PR once we figure out how we want to handle them. This does mean that the 'nan' -> Datetime bug is still present. I could add the check here that discards Datetime as a logical type if the entire column is NaT and that would at least avoid the issue for now.

Reply:
cool, just double checking! Don't need to handle it right now, but maybe it's worth this PR having its own issue outlining the feature change to have fully null columns be Unknown